jetty-ee10/jetty-ee10-maven-plugin/src/main/java/org/eclipse/jetty/ee10/maven/plugin/OverlayManager.java

```diff
@@ -123,17 +127,18 @@ protected List<Overlay> getOverlays()
      * @return the location to which it was unpacked
      * @throws IOException if there is an IO problem
      */
-    protected  Resource unpackOverlay(Overlay overlay)
+    protected Resource unpackOverlay(Overlay overlay)
         throws IOException
     {
         if (overlay.getResource() == null)
             return null; //nothing to unpack
 
         //Get the name of the overlayed war and unpack it to a dir of the
         //same name in the temporary directory
-        String name = overlay.getResource().getFileName();
+        Path p = Paths.get(URIUtil.unwrapContainer(overlay.getResource().getURI()));
```

Comment:
Could you please modify the comment (or maybe add some form of assertion?) to say that the overlay's resource is expected to be of this kind: `jar:file:/some/file.war!/`?